src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescription.cs

```diff
@@ -15,6 +15,22 @@ namespace Microsoft.Identity.Abstractions
     /// </summary>
     public class CredentialDescription
     {
+        private string? _cachedId;
+
+        /// <summary>
+        /// Gets a unique identifier for a CredentialDescription based on <see cref="SourceType"/> and <see cref="ReferenceOrValue"/>.
+        /// </summary>
+        public string Id
```

Comment:
any tests to add?

Reply:
maybe add a theory test around this place?
https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/blob/45cb5a77e02a3b98b2f6d1e11f1c6cb34c83e25b/test/Microsoft.Identity.Abstractions.Tests/CredentialDescriptionTest.cs#L367

Reply:
Adding unit tests...

Reply:
Added unit tests